src/design-system/Pagination.js

```
@@ -0,0 +1,70 @@
+const Pagination = ({ type = 'number', total = 1, current = 1, onChange }) => {
+  if (total < 1) return null;
+
+  if (type === 'dot') {
+    return (
+      <div style={{ display: 'flex', gap: spacing.sm, alignItems: 'center' }}>
+        {Array.from({ length: total }).map((_, idx) => {
+          const isActive = idx + 1 === current;
+          return (
+            <button
+              key={idx}
+              onClick={() => onChange && onChange(idx + 1)}
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)